rust/src/dcerpc/log.rs

```diff
@@ -0,0 +1,45 @@
+/* Copyright (C) 2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+use crate::dcerpc::dcerpc::*;
+use crate::jsonbuilder::{JsonBuilder, JsonError};
+
+fn log_dcerpc_header(js: &mut JsonBuilder, tx: &DCERPCTransaction) -> Result<(), JsonError> {
+    if tx.req_done == true {
+        js.set_string("request", &dcerpc_type_string(tx.req_cmd))?;
+        js.open_object("req")?;
+        js.set_uint("opnum", tx.opnum as u64)?;
```

Comment:
is `opnum` valid for all request types?

Reply:
No, just the standard DCERPC requests. For the rest, it'll be 0. Do you think it should be structured differently? Or maybe a check about the request type before adding opnum to the array?

Reply:
In the SMB/DCE logic the opnum field is only added for REQUEST

Reply:
OK. I'll match again with the SMB?DCE logic and add the missing parts.